logger/logger.go

```diff
@@ -121,11 +124,13 @@ func SetLogger(l Logger) {
 
 type Log struct {
 	log *log.Logger
+
+	debug bool
 }
 
 // Debug only prints if SNAPD_DEBUG is set
 func (l Log) Debug(msg string) {
-	if osutil.GetenvBool("SNAPD_DEBUG") {
+	if l.debug || osutil.GetenvBool("SNAPD_DEBUG") {
```

Comment:
This kind of bugs me, should we just cache GetenvBool result when the logger is constructed? It's not like getting and parsing this all the time is free.

Reply:
We can, I was thinking the same and then decided to keep the change as small as possible. Also I was considering that SNAPD_DEBUG could be set later after the logger was constructed (but that seems to be very much a corner case).